core/src/data/clientGeoJsonSource.cpp

```diff
@@ -0,0 +1,186 @@
+#include "clientGeoJsonSource.h"
+#include "mapbox/geojsonvt/geojsonvt_convert.hpp"
+#include "mapbox/geojsonvt/geojsonvt_types.hpp"
+#include "platform.h"
+#include "util/geom.h"
+
+using namespace mapbox::util;
+
+namespace Tangram {
+
+const double extent = 4096;
+const uint8_t maxZoom = 18;
+const uint8_t indexMaxZoom = 18;
+const uint32_t indexMaxPoints = 100000;
+double tolerance = 0.01;
+
+Point ClientGeoJsonSource::transformPoint(geojsonvt::TilePoint pt) const {
+    return { 2 * pt.x / extent - 1, 1 - 2 * pt.y / extent, 0 };
+}
+
+ClientGeoJsonSource::ClientGeoJsonSource(const std::string& _name, const std::string& _url)
+    : DataSource(_name, _url) {
+
+    if (!_url.empty()) {
+        // Load from file
+        const auto& string = stringFromResource(_url.c_str());
+        addData(string);
+    }
+}
+
+void ClientGeoJsonSource::addData(const std::string& _data) {
+
+    m_features = geojsonvt::GeoJSONVT::convertFeatures(_data);
```

Comment:
Nothing on tangram side, but was looking at GeoJSONVT code (specially convert code), I think there are a bunch of cpp improvement we can do there. We can definitely contribute back to the library once we are done with some high priority tasks.